department-of-veterans-affairs/caseflow
# Pamatya/appeals 17434

Resolves # [APPEALS-17434](https://vajira.max.gov/browse/APPEALS-17434)

### Description
Added sort_order, key and value in Task_Action.json for PO and VISN objects. created a method sort_actions to sort the actions based on task_action's sort_order. If sort order is not present it would return the default.
fixed the spec failure due to the change 

### Acceptance Criteria
- [x] When a user is on the Case Details page in the Program Office queue and clicks the Actions dropdown menu, the task actions should be ordered in the following way:
    1. Mark task in progress
    2. Put task on hold
    3. Assign to VISN
    4. Documents ready for VHA CAMO review
    5. Return to VHA CAMO

- [x] When a user is on the Case Details page in the VISN queue and clicks the Actions dropdown menu, the task actions should be ordered in the following way:
    1. Mark task in progress
    2. Put task on hold
    3. Documents ready for Program Office review
    4. Return to Program Office

### Testing Plan
SETUP
```
program_offices =  VhaProgramOffice.all
program_offices.each do |po|  
 FactoryBot.create_list(:assess_documentation_task, 5 , :assigned, assigned_to: po) 
 FactoryBot.create_list(:assess_documentation_task, 4, :in_progress, assigned_to: po) 
 FactoryBot.create_list(:assess_documentation_task, 5, :on_hold, assigned_to: po) 
 FactoryBot.create_list(:assess_documentation_task, 3, :completed, assigned_to: po)
end

regional_offices =  VhaRegionalOffice.all
regional_offices.each do |ro|  
 FactoryBot.create_list(:assess_documentation_task, 5 , :assigned, assigned_to: ro) 
 FactoryBot.create_list(:assess_documentation_task, 4, :in_progress, assigned_to: ro) 
 FactoryBot.create_list(:assess_documentation_task, 5, :on_hold, assigned_to: ro) 
 FactoryBot.create_list(:assess_documentation_task, 3, :completed, assigned_to: ro)
end
```
1. Switch User to VISNADMIN
2. Go to the Queue -> assigned tab
3.  Click on Veteran's Name to go to the case detail page. 
4. Click on Actions drop down and check if the order is as defined above.
5. Switch User To VHAPOADMIN
6. Repeat Steps from 2-4. 



### User Facing Changes

 BEFORE|AFTER
 ---|---
**VHA-Program office's Actions DropDown**
BEFORE
![image](https://user-images.githubusercontent.com/122557351/234094585-af99b69d-3547-4a33-82a2-a496a2edf2c1.png)


|AFTER

![image](https://user-images.githubusercontent.com/122557351/234092167-7d12e0a7-5cf2-40c3-a276-1622a265754d.png)

**VISN Action's DropDown**
Before 
![image](https://user-images.githubusercontent.com/122557351/234094289-9f1c9a9f-4a0c-48d1-83ca-f590cb807af2.png)

| After
![image](https://user-images.githubusercontent.com/122557351/234092444-bfc56aab-0760-415f-86fb-01b674610b82.png)
